catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/IngestCommand.java

```diff
@@ -503,11 +503,11 @@ private Metacard readMetacard(File file) throws IngestException {
 
   private Metacard generateMetacard(InputStream message) throws IOException {
     try {
-      if (message != null) {
-        return transformer.get().transform(message);
-      } else {
-        throw new IllegalArgumentException("Data file is null.");
+      if (message == null || !transformer.isPresent()) {
+        throw new IllegalArgumentException(
+            "Transformation Failed for transformer: " + transformerId);
       }
+      return transformer.get().transform(message);
```

Comment:
We could also do the following to take advantage of the Optional's provided methods.
```java
Validate.notNull(message); //or a standard (if (message == null) {throw exception})

return transformer
    .map(InputTransformer::transform)
    .orElseThrow(() -> new IllegalArgumentException("Could not get transformer for " + transformerId));
```

Reply:
Good idea. The whole block would look like:

    if (message == null) {
      throw new IllegalArgumentException("Data file is null.");
    }

    try {
      return transformer
              .map(metacard -> metacard.transform(message))
              .orElseThrow(() -> new IllegalArgumentException("Could not get transformer for " + transformerId));
    } catch (Exception e) {
      throw new IllegalArgumentException("Transformation failed for transformer: " + transformerId, e);
    }

But for some reason error catching on the `metacard.transform(message)` doesn't work. Maybe because its in a return statement.

I think it would be better to avoid the headache of that error catch and just stick with what we had.

Reply:
Oh right. Poor support for exceptions within streaming blocks in java. It's because the `Optional::map` takes a Function<T,R> (which does not allow checked exceptions since its not a part of the signature). Oh well ¯\\\_(ツ)\_/¯